### PR TITLE
CaaS 2.4 features

### PR DESCRIPTION
**Pull request by samuelchong**
### Description

Support for CaaS API 2.4
- import customer image 
- clone image (inc. NGOC)
- Add feature exchange nic vlans			
- Add feature to Change NIC network adapter of a server

### Status

Replace this: describe the PR status. Examples:
- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [x] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [ ] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by tonybaloney** [comment]
I assume you branched this from #940? We should close that first.

**Comment by tonybaloney** [comment]
this now needs to be rebased, thanks Sam

**Comment by samuelchong** [comment]
Thx @tonybaloney. Rebase done.

File: libcloud/common/dimensiondata.py [hunk 1]
```diff
@@ -483,7 +483,7 @@ def request_with_orgId_api_1(self, action, params=None, data='',
     def request_with_orgId_api_2(self, action, params=None, data='',
                                  headers=None, method='GET'):
         action = "%s/%s" % (self.get_resource_path_api_2(), action)
-
+        print(action)
```
Comment (tonybaloney) [inline libcloud/common/dimensiondata.py]: please take this debug helper out

File: libcloud/compute/drivers/dimensiondata.py [hunk 2]
```diff
@@ -867,6 +869,110 @@ def list_networks(self, location=None):
...
+
+            if is_guest_os_customization is not None:
+                ET.SubElement(
+                    import_image_elem,
+                    'urn:guestOsCustomization'
+                ).text = is_guest_os_customization
+
+            if len(tagkey_name_value_dictionaries) > 0:
+                for k, v in tagkey_name_value_dictionaries.items():
+                    print(k, v)
```
Comment (tonybaloney) [inline libcloud/compute/drivers/dimensiondata.py]: and this one

Comment (samuelchong) [reply libcloud/compute/drivers/dimensiondata.py]: thx @tonybaloney. Print line removed. What do you think about the idea building a debug mode feature to the framework?

**Comment by tonybaloney** [comment]
:+1: merging, thanks Sam

**Pull request closed without merging**